### DISCUSSION TITLE
Prevent endless loop if root workspace is published package

### DISCUSSION THREAD

**Discussion by luwes** [vercel/turborepo#8485] [Ideas]
### Goals

NPM workspaces allows the root package to be a published package.
This root package can be a dependency of another workspace so it can trigger the originating `build` script again.
There should be a feature that prevents getting in an endless loop calling the originating build script over and over.

This is an example of such a use case https://github.com/muxinc/player.style/blob/main/package.json#L26

### Background

Alternatives are checking an env var in the NPM script or 
removing the NPM scripts in the root workspace and use the turbo CLI directly or
use different NPM script naming but this is easily messed up and cumbersome.

### Proposal

Internally it could be quite simple of checking an environment variable or if the next script is the same script as the original script.

**Comment by anthonyshew** [comment]
What use case does this solve for?

**Reply by luwes** [reply]
The use case is to support a root package that can be published (not private) and has NPM scripts with the same naming than the other workspaces, this root package can be a dependency of other workspaces.

**Reply by anthonyshew** [reply]
I ran `turbo build` in that repository and things went smoothly. What should I be looking for?

**Reply by luwes** [reply]
thanks for checking!
yes, it's when one adds the NPM script `"build": "turbo build"` and run `npm run build` that the endless loop occurs.
generally those NPM scripts are in the root package.

**Reply by anthonyshew** [reply]
Ah, I see. We could probably build something in to help with this. As a workaround, you could use a different name for the script in `./package.json`:

```json
{
  "scripts": {
    "build:all": "turbo run build"
  }
}
```
